### PR TITLE
Add `.reset()` method

### PR DESCRIPTION
**Pull request by husek**
Linked with: https://github.com/sindresorhus/electron-store/pull/84

### CODE REVIEW

**Comment by sindresorhus** [comment]
It should also support resetting to the default value if the user defined the `defaults` option (which should also be documented).

File: index.d.ts [hunk 1]
```diff
@@ -265,6 +265,13 @@ declare class Conf<T = any> implements Iterable<[keyof T, T[keyof T]]> {
 	*/
 	get<K extends keyof T>(key: K, defaultValue?: T[K]): T[K];
 
+	/**
+	Reset an item to its default value (as defined on its schema).
+
+	@param key - The key of the item to get.
```
Comment (sindresorhus) [inline index.d.ts]: The description here is not correct.

**Comment by sindresorhus** [comment]
I think it would be useful if it could reset multiple keys too.

File: index.test-d.ts [hunk 2]
```diff
@@ -62,6 +63,8 @@ conf.set('unicorn', false);
 
 expectType<string>(conf.get('foo'));
 expectType<string>(conf.get('foo', 'bar'));
+conf.reset('foo');
+expectType<string>(conf.get('foo'));
```
Comment (sindresorhus) [inline index.test-d.ts]: This assertion is meaningless, as it's just a type assertion and we already checked `.get`.

File: readme.md [hunk 3]
```diff
@@ -279,6 +279,10 @@ Set multiple items at once.
 
 Get an item or `defaultValue` if the item does not exist.
 
+#### .reset(key)
+
+Reset an item to its predefined `defaultValue`.
```
Comment (sindresorhus) [inline readme.md]: The description here should be the same as in the index.d.ts

File: test.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ const fixture = '🦄';
 
 test.beforeEach(t => {
 	t.context.config = new Conf({cwd: tempy.directory()});
+	t.context.configWithSchema = new Conf({cwd: tempy.directory(), schema: {foo: {default: 42}}});
```
Comment (sindresorhus) [inline test.js]: This should be defined in the test itself as it's only used once.

**Comment by husek** [comment]
> It should also support resetting to the default value if the user defined the `defaults` option (which should also be documented).

Just added support to `defaults`

File: index.d.ts [hunk 5]
```diff
@@ -265,6 +265,13 @@ declare class Conf<T = any> implements Iterable<[keyof T, T[keyof T]]> {
 	*/
 	get<K extends keyof T>(key: K, defaultValue?: T[K]): T[K];
 
+	/**
+	 Reset an item to its default value, as defined on `defaults` or its `schema` option.
+
+	 @param keys - The key/keys of the item/items to reset.
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
	 @param keys - The keys of the items to reset.
```

File: index.d.ts [hunk 6]
```diff
@@ -265,6 +265,13 @@ declare class Conf<T = any> implements Iterable<[keyof T, T[keyof T]]> {
 	*/
 	get<K extends keyof T>(key: K, defaultValue?: T[K]): T[K];
 
+	/**
+	 Reset an item to its default value, as defined on `defaults` or its `schema` option.
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
	 Reset an item to its default value, as defined by the `defaults` or `schema` option.
```

File: index.js [hunk 7]
```diff
@@ -266,6 +281,14 @@ class Conf {
 		return key in this.store;
 	}
 
+	reset(...keys) {
+		keys.forEach(key => {
```
Comment (sindresorhus) [inline index.js]: Use a `for-of` loop.

File: test.js [hunk 8]
```diff
@@ -95,6 +95,23 @@ test('.has()', t => {
 	t.false(t.context.config.has('missing'));
 });
 
+test('.reset()', t => {
+	t.context.configWithSchema = new Conf({cwd: tempy.directory(), schema: {foo: {default: 42}, bar: {default: 99}}});
+	t.context.configWithDefaults = new Conf({cwd: tempy.directory(), defaults: {foo: 42, bar: 99}});
```
Comment (sindresorhus) [inline test.js]: Put the object items on separate lines. It's hard to read when everything is on one line.

File: index.d.ts (hunk 6 above)
Comment (sindresorhus) [reply index.d.ts]: It was implied that the readme text would also be updated.

**Pull request merged**